### DISCUSSION TITLE
Correlation between RAM, buffer size and batch size?

### DISCUSSION THREAD

**Discussion by gpu-pug** [vectordotdev/vector#18587] [Q&A] [answered]
Hi All,
In my current project we recently decided to set-up disk-buffer to increase durability.
Before that we had vector deployed in k8s with 4Gib RAM for every pod, batch.max_bytes ~5Mib and everything worked fine.
Than, we set up disk buffer with buffer.max_size something around size of the RAM. From this moment, seems like vector started leaking memory, i.e. RAM usage started to grow until OOM killer does his job. I've played with configuration and fixed this problem with lowering  batch.max_bytes or with lowering buffer.max_size (in my case to 62% of RAM).
While the problem seems to be solved, it's unclear to me, how these values correlate.
So, how do they correlate?
How to calculate batch and buffer sizes so no to do a lot of experiments and not to break anything often?

**Comment by jszwedko** [comment, marked as answer]
Hi @gpu-pug !

Roughly you can estimate sink memory usage as: max batch size multiplied by concurrent requests.
